src/components/subscriptions/licenses/LicenseManagementTable/index.jsx

```diff
@@ -0,0 +1,304 @@
+  );
+
+  const debouncedSetCurrentPage = debounce(
+    setCurrentPage,
+    DEBOUNCE_TIME_MILLIS,
+  );
+  // Call back function, handles filters and page changes
+  const fetchData = useCallback(
+    (args) => {
+      // pages index from 1 in backend, DataTable component index from 0
+      if (args.pageIndex !== currentPage - 1) {
+        debouncedSetCurrentPage(args.pageIndex + 1);
+      }
+      debouncedUpdateFilters(args.filters);
+    },
+    [currentPage],
+  );
+
+  // Maps user to rows
+  const rows = useMemo(
+    () => users?.results?.map(user => ({
+      id: user.uuid,
+      email: user.userEmail,
+      emailLabel: <span data-hj-suppress>{user.userEmail}</span>,
+      status: user.status,
+      statusBadge: <LicenseManagementUserBadge userStatus={user.status} />,
+      recentAction: userRecentAction(user),
+    })),
+    [users],
+  );
+
+  // Row action button functions
+  const rowRemindOnClick = (remindUser) => {
+    setRemindModal({
+      open: true,
+      users: [remindUser],
+    });
+  };
+  const rowRevokeOnClick = (revokeUser) => {
+    setRevokeModal({
+      open: true,
+      users: [revokeUser],
+    });
+  };
+
+  // Successful action modal callback
+  const onRemindSuccess = () => {
+    forceRefreshUsers();
+    setRemindModal(modalZeroState);
+    addToast(`User${revokeModal.users.length > 1 ? 's' : ''} successfully reminded`);
+  };
+  const onRevokeSuccess = () => {
+    // refresh subscription and user data to get updated revoke count and revoked user list
+    forceRefreshSubscription();
+    forceRefreshUsers();
+    forceRefreshOverview();
+    setRevokeModal(modalZeroState);
+    addToast(`License${revokeModal.users.length > 1 ? 's' : ''} successfully revoked`);
+  };
+
+  // Bulk Action buttons
+  const bulkRemindOnClick = (usersToRemind, allUsersSelected) => {
+    setRemindModal({
+      open: true,
+      users: usersToRemind,
+      allUsersSelected,
+    });
+  };
+  const bulkRevokeOnClick = (usersToRevoke, allUsersSelected) => {
+    setRevokeModal({
+      open: true,
+      users: usersToRevoke,
+      allUsersSelected,
+    });
+  };
+
+  const showSubscriptionZeroStateMessage = subscription.licenses.total === subscription.licenses.unassigned;
+  return (
+    <>
+      {showSubscriptionZeroStateMessage && <SubscriptionZeroStateMessage /> }
+
+      <DataTable
+        showFiltersInSidebar={showFiltersInSidebar}
+        isFilterable
+        manualFilters
+        isSelectable
+        manualSelectColumn={selectColumn}
+        SelectionStatusComponent={DataTable.ControlledSelectionStatus}
+        defaultColumnValues={{ Filter: TextFilter }}
+        isPaginated
+        manualPagination
+        itemCount={users.count}
+        pageCount={users.numPages || 1}
+        // eslint-disable-next-line no-unused-vars
+        tableActions={(i) => {
+          if (!showSubscriptionZeroStateMessage) {
+            return <DownloadCsvButton />;
+          }
+          return <></>;
```

Comment:
nit: Could we use `return null;` here instead of an empty React fragment? Or does it need to return JSX in both cases?